sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -74,12 +75,26 @@ class QueryExecution(
     sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
   }
 
+  lazy val nonRootCommandExecuted: LogicalPlan = analyzed mapChildren { child =>
```

Comment:
I'm a bit confused. Can you define what is non root command?

Reply:
Oh I see. You mean the `Command` not at the top-level node. My first thought is a command not run as root...